doc/BALANCING.md

```diff
@@ -0,0 +1,389 @@
+  balancing wallet. The return collateral will be directed back to the balancing
+  wallet. This option is synonymous with `CollateralUtxosFromWallet
+  (balancingWallet)`.
+* `CollateralUtxosFromWallet Wallet`: Use UTXOs containing only value from the
+  specified wallet. Transactions using these UTXOs will require the signing of
+  the wallet owner for validation. The return collateral will also be sent to
+  this same wallet.
+* `CollateralUtxosFromSet (Set Api.TxOutRef) Wallet`: Utilize UTXOs from the
+  provided set and direct return collaterals to the designated wallet. Note that
+  if any of these UTXOs belong to a script and are selected by the balancing
+  mechanism, validation will fail because only UTXOs controlled by public keys
+  are permissible as collaterals according to ledger rules. Exercise caution
+  with this option as it may involve losing additional information contained
+  within the chosen UTXO during the balancing process.
+
+If auto-balancing is disabled, the collateral calculation will still be based on
+the fee determined by the [`FeePolicy`](#fee-policy). Currently, users cannot
+specify the exact amount of collateral to be used in a transaction, unlike
+fees. This limitation is intentional to minimize user interaction with the
+collateral mechanism, which is rarely needed. Additionally, `cooked-validators`
+does not support issuing new transactions after a validation failure, making the
+collateral mechanism less relevant outside of transaction validation
+requirements.
+
+## Balancing algorithm
+
+The balancing algorithm operates by taking a transaction skeleton as input
+within a `MonadBlockChain` environment and returning this skeleton with
+associated fees, collaterals, and a return collateral wallet. These four
+elements are governed by the [balancing options](#balancing-options). The
+algorithm comprises several components. While not all components will be
+described in detail here (the code is thoroughly commented), we will focus on
+the most important and challenging aspects.
+
+### Reaching a given value with a set of utxos
+
+When balancing a transaction, we start with a set of candidate UTXOs and need to
+decide on a subset that:
+- Is sufficient to cover the missing value in the inputs of the skeleton.
+- Either provides exactly the right amount of funds or enough extra to sustain
+  an additional output in terms of minimal ADA (as the excess will be returned
+  to the balancing wallet).
+- Is minimal, meaning it does not include any UTXOs that do not contribute to
+  reaching the required value. In other words, removing any UTXO from the chosen
+  subset should prevent achieving the required value.
+  
+We also have similar requirements for computing a suitable subset of
+collaterals, with the additional constraint of having a maximum number of
+elements in the subsets.
+  
+The function `reachValue` performs this subset computation. Here is its
```

Comment:
The name `reachValue` might be a bit generic and not very self explanatory. I don't have a better suggestion in mind though.

Reply:
I leave this open for a while in case we find a better name, i don't have any myself.